In [1]:
import pandas as pd


rA = pd.read_csv('realAll.csv')

In [2]:
rA

Unnamed: 0   월    1             0
0             1   1   62           NaN
1             2   1   96           NaN
2             3   1  102           NaN
3             4   1  100        박무\n연무
4             5   1   95        박무\n연무
..          ...  ..  ...           ...
360          27  12   94            연무
361          28  12   94           NaN
362          29  12   78         비\n박무
363          30  12   91  비\n눈\n박무\n연무
364          31  12  106           NaN

[365 rows x 4 columns]

In [4]:
rA = rA.rename(columns={'Unnamed: 0':'날짜'})
rA

날짜   월    1             0
0     1   1   62           NaN
1     2   1   96           NaN
2     3   1  102           NaN
3     4   1  100        박무\n연무
4     5   1   95        박무\n연무
..   ..  ..  ...           ...
360  27  12   94            연무
361  28  12   94           NaN
362  29  12   78         비\n박무
363  30  12   91  비\n눈\n박무\n연무
364  31  12  106           NaN

[365 rows x 4 columns]

In [5]:
rA.to_csv('realAll.csv',index=False) #index = False를 해줘야 값만 보내짐.

In [6]:

rA = pd.read_csv('realAll.csv')
rA

날짜   월    1             0
0     1   1   62           NaN
1     2   1   96           NaN
2     3   1  102           NaN
3     4   1  100        박무\n연무
4     5   1   95        박무\n연무
..   ..  ..  ...           ...
360  27  12   94            연무
361  28  12   94           NaN
362  29  12   78         비\n박무
363  30  12   91  비\n눈\n박무\n연무
364  31  12  106           NaN

[365 rows x 4 columns]

In [7]:
rA = rA.rename(columns={'1':'사고수'})

In [8]:
rA = rA.rename(columns={'0':'날씨'})

In [9]:
rA

날짜   월  사고수            날씨
0     1   1   62           NaN
1     2   1   96           NaN
2     3   1  102           NaN
3     4   1  100        박무\n연무
4     5   1   95        박무\n연무
..   ..  ..  ...           ...
360  27  12   94            연무
361  28  12   94           NaN
362  29  12   78         비\n박무
363  30  12   91  비\n눈\n박무\n연무
364  31  12  106           NaN

[365 rows x 4 columns]

In [10]:
rA.to_csv('realAll.csv',index=False) #index = False를 해줘야 값만 보내짐.

In [11]:

rA = pd.read_csv('realAll.csv')
rA

날짜   월  사고수            날씨
0     1   1   62           NaN
1     2   1   96           NaN
2     3   1  102           NaN
3     4   1  100        박무\n연무
4     5   1   95        박무\n연무
..   ..  ..  ...           ...
360  27  12   94            연무
361  28  12   94           NaN
362  29  12   78         비\n박무
363  30  12   91  비\n눈\n박무\n연무
364  31  12  106           NaN

[365 rows x 4 columns]

In [12]:
rA.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 365 entries, 0 to 364
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   날짜      365 non-null    int64 
 1   월       365 non-null    int64 
 2   사고수     365 non-null    int64 
 3   날씨      235 non-null    object
dtypes: int64(3), object(1)
memory usage: 11.5+ KB


In [13]:
rA.isnull().sum()


날짜       0
월        0
사고수      0
날씨     130
dtype: int64

In [14]:
rA['사고수'].astype(int)
rA.sum()

C:\Users\BIT\AppData\Local\Temp/ipykernel_17068/3549867158.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  rA.sum()


날짜      5738
월       2382
사고수    39258
dtype: int64

In [15]:
rAw = rA.dropna()
rAw['사고수'].astype(int)
rAw.sum()

날짜                                                  3632
월                                                   1369
사고수                                                25063
날씨     박무\n연무박무\n연무연무채운연무연무연무연무박무\n연무눈\n박무\n연무눈눈연무연무연...
dtype: object

전체 = 39258

날씨 있는 날 사고수 = 25063
날씨 없는 날 사고수 = 14195

2019년은 365일. 
그중 날씨가 있는 날은 235일
없는 날은 130 일. 

사고는 날씨와 크게 상관이 없다..

### 오히려 

날씨가 있는 날은 106.65 건으로 날씨가 없는 날보다 조금(2.54건 적다). (25063/235)

날씨가 없는 날 사고가 1일에 평균 109.19건  (14195/130)